src/server/journal/streamer.cc

```diff
@@ -279,7 +280,6 @@ bool RestoreStreamer::ShouldWrite(cluster::SlotId slot_id) const {
 
 void RestoreStreamer::WriteBucket(PrimeTable::bucket_iterator it) {
   if (it.GetVersion() < snapshot_version_) {
-    FiberAtomicGuard fg;
```

Comment:
Untill today we said that bucket serialization is atomic (this is why the fiber gaurd)
Now when you change this assumption we need to make sure no one other fiber is trying to change the keys in this bucket while we are in the middle of serialization.
When a transaction comes to change a value of key it will first check the snapshot version of the bucket if pre update cb is needed to be excuted. Because the flow here is first setting the snapshot version and than can preempt while serializing you can get to a point where you edit the container while you did not finish the serialization.

Reply:
and I think our cluster migration tests are lacking because we do not run seeder while migrating right?

Reply:
> and I think our cluster migration tests are lacking because we do not run seeder while migrating right?

I think this is challenging because of the way our seeder is built(?), in that we need to first capture and then compare, but if we run seeder during migration and something breaks, we can't tell because capture will capture the existing state.
Or is there a way to compare to what we had sent with the seeder?

Reply:
I think we can solve this by running the seeder twice with the same seed number, this should generate the same command sequence (this needs to be verified). This way we can run the seeder again on a new seprate instance and compare the captures between this instance the tested cluster